### PR TITLE
libpcap: add support for dbus, libnl and rdma

### PR DESCRIPTION
**Pull request by valgur**
Build log with all options enabled on Linux: https://gist.github.com/valgur/bb1ff3fd0ca5045a378ddd228c5588b3

### CODE REVIEW

File: recipes/libpcap/all/conanfile.py [hunk 1]
```diff
@@ -105,16 +117,16 @@ def generate(self):
...
-            yes_no = lambda v: "yes" if v else "no"
             tc.configure_args.extend([
-                f"--enable-usb={yes_no(self.options.get_safe('enable_libusb'))}",
-                "--disable-universal",
-                "--without-libnl",
+                "--disable-universal",  # don't build universal binaries on macOS
+                "--enable-usb" if self.options.get_safe("enable_libusb") else "--disable-usb",
+                "--enable-dbus" if self.options.get_safe("enable_dbus") else "--disable-dbus",
+                "--enable-rdma" if self.options.get_safe("enable_rdma") else "--disable-rdma",
+                "--with-libnl" if self.options.get_safe("enable_libnl") else "--without-libnl",
```
Comment (valgur) [inline recipes/libpcap/all/conanfile.py:125]: `--enable-xyz=yes/no` was not supported for most of these options.

**Review by jwillikers — APPROVED** [review]

**Review by AbrilRBS — DISMISSED** [review]
Sorry that this took so long to get reviewed, looks good to me! :)

Thanks a lot for also including a full compilation log, truly appreciated!

**Review by AbrilRBS — DISMISSED** [review]

**Review by uilianries — CHANGES_REQUESTED** [review]
Blocking for further review. Thank you for your update! libpcap is a powerful library!

**Comment by uilianries** [comment]
@valgur I just tried to build locally, but using the official Docker image `conanio/gcc11-ubuntu16.04:2.7.0` and it fails to build:

[libpcap-1.10.4-linux-all-options.log](https://github.com/user-attachments/files/17112240/libpcap-1.10.4-linux-all-options.log)

The error occurred because libpcap was not capable to find those dependencies via pkg-config. The PkgConfigDeps is missing in the recipe. Furthermore, there is a new bug in Conan client that does not allow us renaming the .pc for `libnl::nl-genl` component, so libpcap is not capable to find the expected pc file as well. 

The commit [9271f24](https://github.com/conan-io/conan-center-index/pull/23383/commits/9271f24e12a79e88b8bbb867c837127b6117fa97) fixes the situation by add PkgConfigDeps and renaming the generated .pc file, so we do not patch pcap directly (much cheaper).

[libpcap-1.10.4-linux-pkgconfdeps.log](https://github.com/user-attachments/files/17112323/libpcap-1.10.4-linux-pkgconfdeps.log)

Please, consider using those docker images in the future, when providing the build lob. It did not fail for you, as you system is well configured, having all dependencies installed already.

**Comment by uilianries** [comment]
Depends on https://github.com/conan-io/conan-center-index/pull/25380 first.

It will fix the correct pkg-config missing for libnl when active with libpcap.

**Comment by uilianries** [comment]
Need extra PR: https://github.com/conan-io/conan-center-index/pull/25386

**Comment by valgur** [comment]
LGTM. Thank you for the libnl fix, @uilianries!

**Review by AbrilRBS — APPROVED** [review]

**Review by uilianries — APPROVED** [review]
LGTM. Now libpcap is really consuming Conan dependencies for those new options:

[lipcap-1.10.4-linux-shared.log](https://github.com/user-attachments/files/17159718/lipcap-1.10.4-linux-shared.log)
[lipcap-1.10.4-linux-static.log](https://github.com/user-attachments/files/17159720/lipcap-1.10.4-linux-static.log)

**Pull request merged**